lib/src/DateRangePicker/DateRangePickerInput.tsx

```diff
@@ -81,15 +86,23 @@ export const DateRangePickerInput: React.FC<DateRangeInputProps> = ({
     }
   }, [currentlySelectingRangeEnd, open]);
 
+  // TODO: rethink this approach. We do not need to wait for calendar to be updated to rerender input (looks like freezing)
+  // TODO: so simply break 1 react's commit phase in 2 (first for input and second for calendars) by executing onChange in the next tick
+  const lazyHandleChangeCallback = React.useCallback(
```

Comment:
Described more in this [twitter thread](https://twitter.com/dmtrKovalenko/status/1250777792930873349)

Reply:
Is the underlying issue that the calendar (containing ~30 days instances) too slow to update?

Reply:
Yeah it is slow, cause actually 70+ days for standard desktop calendar. 

Moreover it is working like throttle but on the react level. User don’t expect to see January of 0002 year when input is 01/01/2